dask/dataframe/core.py

```diff
@@ -1669,7 +1669,20 @@ def isna(self):
 
     @derived_from(pd.DataFrame)
     def isin(self, values):
-        return elemwise(M.isin, self, list(values))
+        if is_dataframe_like(self._meta):
+            # DataFrame.isin does weird alignment stuff
+            bad_types = (_Frame, pd.Series, pd.DataFrame)
+        else:
+            bad_types = (_Frame,)
+        if isinstance(values, bad_types):
+            raise NotImplementedError(
+                "Passing a %r to `isin`" % typename(type(values))
```

Comment:
I'm a little bit surprised that we can't call `df.isin(my_pandas_series)`.  Can you help explain what was going on here?

Reply:
For `pd.Series.isin` all arguments are treated the same, but for dataframes Pandas does a weird with alignment along the index that we can't really support (see the comment above here).

Reply:
I think that it would be good to guide the user here a bit towards correct behavior.  "Ok, you've told me I can't pass in a pandas series, what can I pass in?"

Or maybe, if they're passing in a series maybe we can coerce to a numpy array for them?

Reply:
> Or maybe, if they're passing in a series maybe we can coerce to a numpy array for them?

That wouldn't match pandas then.